src/FakeItEasy/Creation/ProxyGeneratorResult.cs

```diff
@@ -66,23 +66,22 @@ public ProxyGeneratorResult(object generatedProxy)
         {
             Guard.AgainstNull(generatedProxy, nameof(generatedProxy));
 
-            this.ProxyWasSuccessfullyGenerated = true;
             this.GeneratedProxy = generatedProxy;
         }
 
         /// <summary>
         /// Gets a value indicating whether the proxy was successfully created.
         /// </summary>
-        public bool ProxyWasSuccessfullyGenerated { get; }
+        public bool ProxyWasSuccessfullyGenerated => this.GeneratedProxy is object;

```

Comment:
I thought about something the other day: we *could* use `is notnull` instead of `is object`! We'd just need a `using notnull = System.Object;` in every file... Of course, it could lead to weird things, such as `notnull.Equals(...)` or `new notnull()`, but I like the idea anyway ;)

Reply:
That was the idea